### PR TITLE
chore: Only run Java build on PRs with backend changes

### PR DESCRIPTION
**Pull request by liuverta**
## Impact and Context

So I don't have to wait for this check on client PRs.

## Risks and Area of Effect
- [ ] Is this a breaking change?

I chose these paths by checking which files and directories at [the repo root](https://github.com/VertaAI/modeldb) were changed in the last few months. Hopefully I captured everything relevant to not break the check for Java PRs.

## Testing
- [ ] Unit test
- [ ] Deployed to dev env
- [ ] Other (explain) 

—

## Reverting
- [ ] Contains Migration - _Do Not Revert_

Revert this PR.

### CODE REVIEW

**Review by gustavoteixeirah — APPROVED** [review]

**Review by conradoverta — COMMENTED** [review]
Also changes to `.github/workflows/java-build-push-test.yaml` so that changes to the pipeline run the pipeline?

File: .github/workflows/java-build-push-test.yaml [hunk 1]
```diff
@@ -3,6 +3,11 @@ name: Java Build, Push, Test
 on:
   pull_request:
     types: [opened, reopened, synchronize]
+    paths:
+      - .run
```
Comment (jkwatson-verta) [inline .github/workflows/java-build-push-test.yaml]: this isn't needed...that's just a folder for IDEA run configurations
```suggestion
```

File: .github/workflows/java-build-push-test.yaml [hunk 2]
```diff
@@ -3,6 +3,11 @@ name: Java Build, Push, Test
...
+      - backend
+      - protos
+      - pom.xml
```
Comment (jkwatson-verta) [inline .github/workflows/java-build-push-test.yaml:10]:
```suggestion
      - pom.xml
      - build_and_publish_backend.sh
```

**Review by jkwatson-verta — APPROVED** [review]
Thanks!

**Pull request merged**